File: process_user_videos.py
import json
import os

RAW_PATH = 'raw_videos.json'
VIDEOS_PATH = 'src/data/videos.json'

def process_videos():
    try:
        print(f'Reading raw videos from: {RAW_PATH}')
        with open(RAW_PATH, 'r', encoding='utf-8') as f:
            videos = json.load(f)

        processed_videos = []
        seen_ids = set()

        for video in videos:
            # Clean strings
            for key, value in video.items():
                if isinstance(value, str):
                    video[key] = value.replace('\n', '').strip()

            vid_id = video.get('id')

            # Filter out placeholders
            if vid_id.startswith('SAMPLE'):
                print(f'Skipping placeholder {vid_id}')
                continue

            # Deduplicate
            if vid_id in seen_ids:
                print(f'Skipping duplicate {vid_id}')
                continue

            # Handle blocked embed for TED-Ed
            if vid_id == 'gqFOi9bQJ9c':
                video['embedUrl'] = None
                video['openOnYoutube'] = True
            else:
                video['openOnYoutube'] = False
            
            seen_ids.add(vid_id)
            processed_videos.append(video)

        print(f'Writing {len(processed_videos)} videos to {VIDEOS_PATH}')
        with open(VIDEOS_PATH, 'w', encoding='utf-8') as f:
            json.dump(processed_videos, f, indent=4)
        print('Done.')

    except Exception as e:
        print(f'Error processing videos: {e}')

if __name__ == "__main__":
    process_videos()

File: process_user_videos.js
const fs = require('fs');
const path = require('path');

const RAW_PATH = path.join(process.cwd(), 'raw_videos.json');
const VIDEOS_PATH = path.join(process.cwd(), 'src/data/videos.json');

const processVideos = () => {
    try {
        console.log('Reading raw videos from:', RAW_PATH);
        const rawData = fs.readFileSync(RAW_PATH, 'utf8');
        let videos = JSON.parse(rawData);

        const processedVideos = [];
        const seenIds = new Set();

        for (const video of videos) {
            // Clean strings (remove newlines and trim)
            for (const key in video) {
                if (typeof video[key] === 'string') {
                    video[key] = video[key].replace(/\n/g, '').trim();
                }
            }

            const id = video.id;

            // Filter out placeholders
            if (id.startsWith('SAMPLE')) {
                console.log(`Skipping placeholder ${id}`);
                continue;
            }

            // Deduplicate
            if (seenIds.has(id)) {
                console.log(`Skipping duplicate ${id}`);
                continue;
            }

            // Ensure required fields
            if (!video.tags) video.tags = [];

            // Set embedUrl based on ID (assuming all non-placeholders are embeddable unless known otherwise)
            // The user provided embedUrls, so we keep them, but cleaned.
            // However, I previously identified 'gqFOi9bQJ9c' (TED-Ed) as blocked.
            // I should probably respect my previous finding if it's in the list.
            if (id === 'gqFOi9bQJ9c') {
                video.embedUrl = null; // Block embed for TED-Ed
                video.openOnYoutube = true;
            } else {
                // Ensure embedUrl is set if missing (though user provided it)
                if (!video.embedUrl) {
                    video.embedUrl = `https://www.youtube.com/embed/${id}`;
                }
                video.openOnYoutube = false;
            }

            seenIds.add(id);
            processedVideos.push(video);
        }

        console.log(`Writing ${processedVideos.length} videos to ${VIDEOS_PATH}`);
        fs.writeFileSync(VIDEOS_PATH, JSON.stringify(processedVideos, null, 4));
        console.log('Done.');

    } catch (error) {
        console.error('Error processing videos:', error);
    }
};

processVideos();
